model/ipsec_tunnel.rb

```diff
@@ -7,4 +7,58 @@ class IpsecTunnel < Sequel::Model
   many_to_one :dst_nic, key: :dst_nic_id, class: Nic
 
   include ResourceMethods
+
+  def refresh
+    create_ipsec_tunnel
+    create_private_routes
+  end
+
+  def create_ipsec_tunnel
+    src_namespace = vm_name(src_nic)
+    dst_namespace = vm_name(dst_nic)
+    src_clover_ephemeral = subdivide_network(src_nic.vm.ephemeral_net6)
+    dst_clover_ephemeral = subdivide_network(dst_nic.vm.ephemeral_net6)
+    src_private_addr_6 = src_nic.private_ipv6.to_s.shellescape
+    dst_private_addr_6 = dst_nic.private_ipv6.to_s.shellescape
+    src_private_addr_4 = src_nic.private_ipv4.to_s.shellescape
+    dst_private_addr_4 = dst_nic.private_ipv4.to_s.shellescape
+    src_direction = "out"
+    dst_direction = "fwd"
+
+    spi = "0x" + SecureRandom.bytes(4).unpack1("H*")
+    spi4 = "0x" + SecureRandom.bytes(4).unpack1("H*")
+    key = src_nic.encryption_key
+
+    # setup source ipsec tunnels
+    src_nic.vm.vm_host.sshable.cmd("sudo bin/setup-ipsec " \
+      "#{src_namespace} #{src_clover_ephemeral} " \
+      "#{dst_clover_ephemeral} #{src_private_addr_6} " \
+      "#{dst_private_addr_6} #{src_private_addr_4} " \
+      "#{dst_private_addr_4} #{src_direction} " \
+      "#{spi} #{spi4} #{key}")
+
+    # setup destination ipsec tunnels
+    dst_nic.vm.vm_host.sshable.cmd("sudo bin/setup-ipsec " \
+      "#{dst_namespace} #{src_clover_ephemeral} " \
+      "#{dst_clover_ephemeral} #{src_private_addr_6} " \
+      "#{dst_private_addr_6} #{src_private_addr_4} " \
+      "#{dst_private_addr_4} #{dst_direction} " \
+      "#{spi} #{spi4} #{key}")
```

Comment:
I find it interesting that bilateral set-up (where we must shell into one machine and then the other in one shot) survives another day. I don't have deep thoughts about it yet, except that it requires deep thought.